bin/codebasin

```diff
@@ -220,6 +242,32 @@ def main():
             filtered_platforms=filtered_platforms,
         )
 
+    # Load the analysis file if it exists.
+    if args.analysis_file is not None:
+        path = os.path.realpath(args.analysis_file)
+        if os.path.exists(path):
+            if not os.path.splitext(path)[1] == ".toml":
+                raise RuntimeError(f"Analysis file {p} must end in .toml.")
+
+        with util.safe_open_read_nofollow(path, "rb") as f:
+            try:
+                analysis_toml = util._load_toml(f, "analysis")
+            except BaseException:
```

Comment:
I wanted to originally comment on this but couldn't resolve `util._load_toml`. I think the function should throw `ValueError` instead of having to intercept a `BaseException`

Reply:
I don't really disagree, but I think a fix may be outside of the scope of this PR.

Some of the functions in `util` currently do raise these sorts of errors, but they're not very user-friendly.  Letting one of the exceptions escape would result in an error like "JSON failed schema validation" and some `jsonschema` errors, which wouldn't make sense to somebody who provided a TOML file.

Would you be okay with us opening an issue to track this for 2.0.0?  When we don't have to worry about the behavior of the other `util` functions, we can replace this with a nicer validation interface.

Reply:
Opened as #67.